internal/testhelpers/e2e/session.go

```diff
@@ -332,6 +332,14 @@ func (s *Session) PrepareCommitIdFile(commitID string) {
 	require.NoError(s.T, pjfile.SetLegacyCommit(commitID))
 }
 
+// CommitID is used to grab the current commit ID for the project in our working directory. This is distinct
+// from localcommit.Get() and pjfile.LegacyCommitID() in that it always produces a fresh result from the on-disk file.
```

Comment:
Does this mean we shouldn't be using `localcommit.Get()` or `pjfile.LegacyCommitID()` from integration tests, that we should only be using this? It might be good to add that to this comment. Otherwise, it's not clear to me why we should be using this function.